indra/newview/scripts/lua/LLChatListener.lua

```diff
@@ -0,0 +1,41 @@
+local fiber = require 'fiber'
+local inspect = require 'inspect'
+
+local LLChatListener = {}
+local waitfor = {}
+
+function LLChatListener:new()
+    local obj = setmetatable({}, self)
+    self.__index = self
+    obj.name = 'Chat_listener'
+
+    return obj
+end
+
+function LLChatListener:handleMessages(event_data)
+    --print(inspect(event_data))
+    return true
+end
+
+function LLChatListener:start()
+    waitfor = leap.WaitFor:new(-1, self.name)
```

Comment:
The fact that `leap.WaitFor` resolves without a `local leap = require 'leap'` statement means I must have inadvertently tossed `leap` into the global namespace. I'd still rather see the `require` statement in this module.

Reply:
Agree